app/lib/service/download_counts/compute_30_days_total_counts.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'dart:async';
+import 'dart:math';
+
+import 'package:gcloud/storage.dart';
+import 'package:pub_dev/service/download_counts/backend.dart';
+import 'package:pub_dev/service/download_counts/models.dart';
+import 'package:pub_dev/shared/configuration.dart';
+import 'package:pub_dev/shared/storage.dart';
+import 'package:pub_dev/shared/utils.dart';
+
+Future<void> compute30DaysTotalTask() async {
+  final allDownloadCounts = await downloadCountsBackend.listAllDownloadCounts();
+  final totals = await compute30DayTotals(allDownloadCounts);
+  await updload30DaysTotal(totals);
+}
+
+Future<Map<String, int>> compute30DayTotals(
+    Stream<DownloadCounts> downloadCounts) async {
+  final res = <String, int>{};
+  await for (final dc in downloadCounts) {
+    res[dc.package] = compute30DayTotal(dc);
+  }
+
+  return res;
+}
```

Comment:
Why not merge these with `listAllDownloadCounts`, and make a single method?

Passing around streams is a bit scary, even if I think our stream will retry individual pages as it fetches the entire result.

Reply:
I think logically this is the separation that makes sense....